lib/connextclient/ethprovider.ts

```diff
@@ -0,0 +1,60 @@
+import { Observable, from, combineLatest } from 'rxjs';
+import { ethers, BigNumber } from 'ethers';
+import { mergeMap } from 'rxjs/operators';
+
+type OnChainTransaction = {
+  nonce: number;
+  gasPrice: BigNumber;
+  gasLimit: BigNumber;
+  to: string;
+  value: BigNumber;
+  data: string;
+  chainId: number;
+  v: number;
+  r: string;
+  s: string;
+  from: string;
+  hash: string;
+};
+
+const onChainSendERC20 = (
+  host: string,
+  port: number,
+  chainId: number,
+  seed: string,
+  contractAddress: string,
+  destinationAddress: string,
+  units: string,
+): Observable<OnChainTransaction> => {
+  const provider = new ethers.providers.JsonRpcProvider(
+    {
+      url: `http://${host}:${port}/ethprovider/${chainId}`,
+    },
+    {
+      name: 'simnet',
+      chainId,
+    },
+  );
+  const erc20abi = ['function balanceOf(address) view returns (uint)', 'function transfer(address to, uint amount)'];
+  const signer = ethers.Wallet.fromMnemonic(seed).connect(provider);
+  const erc20 = new ethers.Contract(contractAddress, erc20abi, signer);
+  // convert promises to observables
+  const erc20balance$ = from(erc20.balanceOf(signer.address)) as Observable<BigNumber>;
+  const ethBalance$ = from(provider.getBalance(signer.address));
```

Comment:
Why do we need the Ether balance for sending ERC20 tokens?

Reply:
This in preparation for adding balance verification logic that currently resides in `ConnextClient.withdraw`.

Reply:
I went ahead and removed it right now to keep things easier. The checks are being done before this function is called.